src/atomdb/redis_mongodb/RedisMongoDBAPITypes.cc

```
@@ -145,26 +145,64 @@ unsigned int MongodbDocument::get_size(const string& array_key) {
     return ((*this->document)[array_key]).get_array().value.length() / HANDLE_HASH_SIZE;
 }
 
+bsoncxx::v_noabi::document::view MongodbDocument::get_object(const string& key) {
+    return ((*this->document)[key]).get_document().value;
+}
+
 bsoncxx::v_noabi::document::value MongodbDocument::value() { return this->document.value(); }
 
-void append_custom_attributes(bsoncxx::builder::basic::document& doc,
-                              const Properties& custom_attributes) {
-    for (const auto& [key, value] : custom_attributes) {
-        visit(
-            [&](auto&& arg) {
-                using T = decay_t<decltype(arg)>;
-                if constexpr (is_same_v<T, string>) {
-                    doc.append(bsoncxx::builder::basic::kvp(key, arg));
-                } else if constexpr (is_same_v<T, long>) {
-                    doc.append(bsoncxx::builder::basic::kvp(key, static_cast<int64_t>(arg)));
-                } else if constexpr (is_same_v<T, double>) {
-                    doc.append(bsoncxx::builder::basic::kvp(key, arg));
-                } else if constexpr (is_same_v<T, bool>) {
-                    doc.append(bsoncxx::builder::basic::kvp(key, arg));
-                }
-            },
-            value);
+void add_custom_attributes(bsoncxx::builder::basic::document& doc, const Properties& custom_attributes) {
+    if (!custom_attributes.empty()) {
+        bsoncxx::builder::basic::document custom_attributes_doc;
+        for (const auto& [key, value] : custom_attributes) {
+            visit(
+                [&](auto&& arg) {
+                    using T = decay_t<decltype(arg)>;
+                    if constexpr (is_same_v<T, string>) {
+                        custom_attributes_doc.append(bsoncxx::builder::basic::kvp(key, arg));
+                    } else if constexpr (is_same_v<T, unsigned int>) {
+                        custom_attributes_doc.append(
+                            bsoncxx::builder::basic::kvp(key, static_cast<int64_t>(arg)));
```

If mongoldb doesn't support the difference between long and unsigned int we should rather raise an exception here instead of silently storing a unsigned int as long.

The problem with this silent approach is that user's code will break after they get an atom from AtomDB and try to get the value of the unsigned int property because they will do something like get<unsigned int>(key). It's (by far) preferable that we raise the exception when the user try to insert the atom in AtomDB than letting him do it just to raise the exception latter. Note that a user could populate an. atom DB to be used by someone else.